frontend/src/app/models/base-filter.ts

```diff
@@ -112,27 +115,38 @@ export abstract class BaseFilter<InitialParameters, FilterData> implements Filte
      * already. Setting the filter to default data will deactivate it.
      */
     set(data: FilterData) {
+        this.latestDataPush$.next(data);
         if (!_.isEqual(data, this.currentData)) {
-            this.data.next(data);
-
-            const active = this.active.value;
             const toDefault = this.isDefault(data);
-            const deactivate = active && toDefault;
-            const activate = !active;
-
-            if (activate) {
-                this.activate();
-            } else if (deactivate) {
-                this.deactivate();
-            } else if (active) {
-                this.update.next();
+
+            if (toDefault) {
```

Comment:
```suggestion
            if (this.isDefault(data) {
```
Not assigning this to a variable first is more readable IMO.